src/components/navbars/ProfileNav.jsx

```diff
@@ -1,14 +1,47 @@
-import { Link } from "react-router-dom";
+import { Link } from 'react-router-dom';
+import { logout } from '../../util/DataBaseRequests';
 import logoutIcon from '../../assets/icons/logout.png';
+import { useAuth } from '../../AuthProvider';
+import { useNavigate } from 'react-router-dom';
+import { useState } from 'react';
 
 const ProfileNav = () => {
-    return(
-            <nav className="sm:hidden relative flex flex-row my-4 gap-2 items-center">
-                <div id="user-photo" className="h-10 w-10 rounded-full bg-lightGreen"></div>
-                <p className="font-spartan text-lg font-semibold">User name</p>
-                <Link to="/"><img src={logoutIcon} className="h-6" alt="logout icon"/></Link>
-            </nav>
-    )
+  const [logoutError, setLogoutError] = useState({});
+  const { clearUserSession, userData } = useAuth();
+
+  const navigate = useNavigate();
+  const handleLogout = async () => {
+    try {
```

Comment:
Stretch goal: Adding loading state to communicate to the user that an action is pending. This is something that can be added towards the end if we have time but keep in mind where loading state can be beneficial for user interaction.

Reply:
What is better to add a spinner or just text?

Reply:
Depends on the style you want. Oksana mentioned that it's already a stretch goal. We can talk about it during the standup and we'll make a ticket for it.